### ISSUE TITLE
Deprecation Warnings from SciPy Arraytools in hysteresis_correct function (Anaconda Python 3.5)

### ISSUE THREAD

**Issue by N00bDaan** [gb119/Stoner-PythonCode#7]
Hi,

I'm getting the following VisibleDeprecationWarning

`..\Anaconda3\lib\site-packages\scipy\signal\_arraytools.py:45: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  b = a[a_slice]`

Resulting in:
```
---------------------------------------------------------------------------
TypeError Traceback (most recent call last)
<ipython-input-4-7eb4d939af23> in <module>()
----> 1 hysteresis_correct('QD-SQUID-VSM.dat',correct_background=False,correct_H=False)

..\Anaconda3\lib\site-packages\stoner-0.6.0rc4-py3.5.egg\Stoner\Util.py in hysteresis_correct(data, **kargs)
    362
    363     while True:
--> 364         up,down=_up_down(data)
    365         mx = max(data.x) * (1 - saturation_fraction)
    366         mix = min(data.x) * (1 - saturation_fraction)

..\Anaconda3\lib\site-packages\stoner-0.6.0rc4-py3.5.egg\Stoner\Util.py in _up_down(data)
     30         (Data, Data): Tuple of two DataFile like instances for the rising and falling data.
     31     """
---> 32     f=split_up_down(data)
     33
     34     ret=[None,None]

..\Anaconda3\lib\site-packages\stoner-0.6.0rc4-py3.5.egg\Stoner\Util.py in split_up_down(data, col, folder)
    171     if width % 2 == 0:  # Ensure the window for Satvisky Golay filter is odd
    172         width += 1
--> 173     peaks = list(a.peaks(col, width,xcol=False, peaks=True, troughs=False))
    174     troughs = list(a.peaks(col, width, xcol=False, peaks=False, troughs=True))
    175     if len(peaks) > 0 and len(troughs) > 0:  #Ok more than up down here

..\Anaconda3\lib\site-packages\stoner-0.6.0rc4-py3.5.egg\Stoner\Analysis.py in peaks(self, ycol, width, significance, xcol, peaks, troughs, poly, sort, modify)
   1567             width = len(self) / 20
   1568         assert poly >= 2, "poly must be at least 2nd order in peaks for checking for significance of peak or trough"
-> 1569         d1 = self.SG_Filter(ycol, width, poly, 1)
   1570         d2 = self.SG_Filter(ycol, 2*width, poly, 2) # 2nd differential requires more smoothing
   1571

..\Anaconda3\lib\site-packages\stoner-0.6.0rc4-py3.5.egg\Stoner\Analysis.py in SG_Filter(self, col, points, poly, order, result, replace, header)
    276         else:
    277             data = self.column(col)
--> 278             r = savgol_filter(data, window_length=points, polyorder=poly, deriv=order, mode="interp")
    279         if result is not None:
    280             if not isinstance(header, string_types):

..\Anaconda3\lib\site-packages\scipy\signal\_savitzky_golay.py in savgol_filter(x,window_length, polyorder, deriv, delta, axis, mode, cval)
    342         # the last `window_length` elements.
    343         y = convolve1d(x, coeffs, axis=axis, mode="constant")
--> 344         _fit_edges_polyfit(x, window_length, polyorder, deriv, delta, axis, y)
    345     else:
    346         # Any mode other than 'interp' is passed on to ndimage.convolve1d.

..\Anaconda3\lib\site-packages\scipy\signal\_savitzky_golay.py in _fit_edges_polyfit(x, window_length, polyorder, deriv, delta, axis, y)
    220     halflen = window_length // 2
    221     _fit_edge(x, 0, window_length, 0, halflen, axis,
--> 222               polyorder, deriv, delta, y)
    223     n = x.shape[axis]
    224     _fit_edge(x, n - window_length, n, n - halflen, n, axis,

..\Anaconda3\lib\site-packages\scipy\signal\_savitzky_golay.py in _fit_edge(x, window_start, window_stop, interp_start, interp_stop, axis, polyorder, deriv, delta, y)
    190     # where '-1' is the same as in xx_edge.
    191     poly_coeffs = np.polyfit(np.arange(0, window_stop - window_start),
--> 192                              xx_edge, polyorder)
    193
    194     if deriv > 0:

..\Anaconda3\lib\site-packages\numpy\lib\polynomial.py in polyfit(x, y, deg, rcond, full, w, cov)
--> 563         raise TypeError("expected x and y to have same length")
    568
    569     # set rcond

TypeError: expected x and y to have same length
```

This is due the use of floats for the `window_start` and `window_stop` variables in `..\Anaconda3\lib\site-packages\scipy\signal\_savitzky_golay.py` function `_fit_edge`

When I try to resolve this by making the window limits integers more (similar) errors pop up.

I'm running `Python 3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)]`

Please let me know if I can be of any help fixing this.

**Comment by gb119** [comment]
Thanks for the report - I've seen something like this before, and thought I'd fixed it - but probably only in master rather than a released package (we're still laggards on Python 2.7.x so fixes for Python 3.5 tend to get a bit lost). Right now master is a bit of work in progress, so I'll go and investigate the stable branch on Python 3.5

**Comment by N00bDaan** [comment]
Thanks for the comment; over the weekend tried it on my laptop (running 2.7) and indeed there it ran no problems. :-)

**Comment by gb119** [comment]
I've released a version 0.6.4 which backports a fix for these problems from the master branch and which passes the (rather minimal) unit tests on both 3.5 and 2.7 (one of which does now test the hysteresis correct function). I'll close this issue now, but feel free to open a new one if there are more problems.
